test/utils/bytecode.hpp

```diff
@@ -503,6 +503,58 @@ inline call_instruction<OP_CALLCODE> callcode(bytecode address)
     return call_instruction<OP_CALLCODE>{std::move(address)};
 }
 
+template <Opcode kind>
+struct extcall_instruction
+{
+private:
+    bytecode m_address = 0;
+    bytecode m_value = 0;
+    bytecode m_input = 0;
+    bytecode m_input_size = 0;
+
+public:
+    explicit extcall_instruction(bytecode address) : m_address{std::move(address)} {}
+
+
+    template <Opcode k = kind>
+    typename std::enable_if<k == OP_EXTCALL, extcall_instruction&>::type value(bytecode v)
```

Comment:
You can use C++20's `requires(k == OP_EXTCALL)`.

Reply:
:eyes: presuming you've meant 

```
    template <Opcode k = kind>
        requires requires { k == OP_EXTCALL; }
    extcall_instruction& value(bytecode v)
```

as this compiles and works. I'll leave deciphering the docs for `requires` for later :joy: Crazy stuff